src/components/MyAccount/NewSettings/UsernameForm.tsx

```diff
@@ -0,0 +1,216 @@
+}
+
+interface UsernameFormProps {
+  patron: Patron
+  usernameState
+}
+
+const UsernameForm = ({ patron, usernameState }: UsernameFormProps) => {
+  const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isEditing, setIsEditing] = useState(false)
+  const [error, setError] = useState(false)
+  const [input, setInput] = useState(patron.username || "")
+  const [tempInput, setTempInput] = useState(input)
+
+  const { setUsernameStatus, setUsernameStatusMessage } = usernameState
+
+  const validateUsername = (username: string | null) => {
+    if (username === null) return false
+    if (username === "") return true
+    const usernameRegex = /^[a-zA-Z0-9]{5,15}$/
+    return usernameRegex.test(username)
+  }
+
+  const cancelEditing = () => {
+    setTempInput(input)
+    setIsEditing(false)
+    setError(false)
+  }
+
+  const handleInputChange = (e) => {
+    const { value } = e.target
+    setTempInput(value)
+    if (!validateUsername(value)) {
+      setError(true)
+    } else {
+      setError(false)
+    }
+  }
+
+  const submitInput = async () => {
+    setIsLoading(true)
+    setIsEditing(false)
+    setUsernameStatus("")
+    const submissionInput = tempInput === null ? "" : tempInput
+    try {
+      const response = await fetch(
+        `${BASE_URL}/api/account/username/${patron.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ username: submissionInput }),
+        }
+      )
+      const responseMessage = await response.json()
+      if (responseMessage !== "Username taken" && response.status === 200) {
+        await getMostUpdatedSierraAccountData()
+        setUsernameStatus("success")
+        setInput(submissionInput)
+        setTempInput(submissionInput)
+      } else {
+        setUsernameStatus("failure")
+        setUsernameStatusMessage(
+          responseMessage === "Username taken"
+            ? usernameStatusMessages.USERNAME_FAILURE
+            : usernameStatusMessages.FAILURE
+        )
+        setTempInput(input)
+      }
+    } catch (error) {
+      setUsernameStatus("failure")
+      setUsernameStatusMessage(usernameStatusMessages.FAILURE)
+      console.error("Error submitting username:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const editUsernameField = (
+    <>
+      <Flex flexDir="row" alignItems="flex-start">
+        <TextInput
+          sx={{
+            width: { base: "87%", md: "300px" },
+            display: "flex",
+            flexDirection: "column-reverse",
+            label: {
+              fontWeight: "400",
+              color: error ? "ui.error.primary" : "ui.black",
+            },
+          }}
+          value={tempInput || ""}
+          id="username-input"
+          labelText="Must be 5-15 characters and use only letters (a-z) and numbers (0-9)"
```

Comment:
The space between the input and label seems too small, right?

Reply:
Yes, j added more styling

Reply:
It's like a mix of the label and helper text on this component I'm not sure how design arrived at this

Reply:
If there is a default that makes sense functionally (like the helper text) coming from reservoir that deviates from the designs in a small way, can you ask Apoorva if it's alright to use the default instead of overriding?